docs/design/design-patterns/rest-api-design-guidance/creating-api-contract.md

```diff
@@ -0,0 +1,56 @@
+# Creating API Contract
+
+As different development teams exposes APIs to access various REST based services, its important to have an API contract to share between producer and consumers of APIs. [Open API](https://www.openapis.org/) format is one of the most popular API description format. This Open API document can be produced in two ways:
+
+- Design-First - Team starts developing APIs by first describing API designs as an Open API document and later generates server side boilerplate code with the help of this document.
+- Code-First - Team starts writing the server side api interface code e.g. controllers, DTOs etc. and later generates and Open API document from it.
+
+## Design-First Approach
+
+A Design-First approach means that APIs are treated as "first-class citizens" and everything about a project revolves around the idea that at the end these APIs will be consumed by clients. So based on the business requirements api development team first start describing API designs as an Open API document and collaborate with the stakeholders to gather feedback.
```

Comment:
NIT: have noticed that in some places API is capitalized but this is not consistent throughout the document, suggest reviewing to make it consistent everywhere.

Reply:
Updated it. Thank you.